dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/ZipFileContainer.java

```diff
@@ -902,6 +921,25 @@ public int locatePath(String r_path) {
         }
         return location;
     }
+    
+    @Trivial
+    private int locateMultiReleasePath(String path) {
+        // Multi-release files may be at /com/foo/A.class and /META-INF/versions/9/com/foo/A.class
+        // Since we commonly bindly try-load META-INF and java.* entries, optimize those out
+        if (path == null || !path.endsWith(".class") || path.startsWith("META-INF") || path.startsWith("java/")) {
+            return -1;
+        }
+        for (int currentVersion = CURRENT_JAVA_VERSION; currentVersion >= 8; currentVersion--) {
```

Comment:
This concept was carried over from `JarFile` in OpenJDK, but in OpenJDK you can create a `JarFile` at a specific release using the new `new JarFile(...., Runtime.Version)` ctor. In that case it would be possible to make a `JarFile` that uses JDK 8 code when you are running on JDK 11 for example. However, in Liberty we don't do that and we'll just always resolve to the current runtime version, so you are right, the `versions/8/` folder could never be accessed.